conans/test/integration/toolchains/cmake/test_cmaketoolchain.py

```diff
@@ -784,3 +784,27 @@ def test_pkg_config_block():
     pkg_config_path_set = 'set(ENV{PKG_CONFIG_PATH} "%s$ENV{PKG_CONFIG_PATH}")' % \
                           (client.current_folder.replace("\\", "/") + pathsep)
     assert pkg_config_path_set in toolchain
+
+
+def test_set_cmake_lang_compilers():
+    profile = textwrap.dedent(r"""
+    [settings]
+    os=Windows
+    arch=x86_64
+    compiler=clang
+    compiler.version=15
+    compiler.libcxx=libstdc++11
+    [conf]
+    tools.cmake.cmaketoolchain:compilers = [("C", "/my/local/gcc"), ("cxx", "C:\\local\\g++"),]
```

Comment:
It would be better to have a ``c_compiler`` and ``cxx_compiler`` confs, no need for this tuple trick, and C, C++ compilers deserve their own named conf.

My main doubt is if this would be a ``tools.build:cxx_compiler`` generic one instead of a CMake specific one, so other generators could also leverage them (probably defining directly the CC/CXX env-var?)

Reply:
I see your point, but I think they should be 5:

```
C
CXX
OBJC (Objective C)
OBJCXX (Objective C++)
RC (Windows Resource Compiler)
```

Reply:
I am fine with having 5 properties. Maybe it would be more idiomatic to have a dict instead? Maybe it is the list of tuples that is hurting a bit the UI?

Reply:
I like the idea of it being a dictionary instead of adding the 5 configs, but if we add them I would probably also add _CUDA_.

Reply:
In my opinion, I prefer having five properties if it's alright for you all.